cylc/flow/cycling/loader.py

```diff
@@ -63,8 +63,16 @@ class DefaultCycler:
     TYPE: str
 
 
+@overload
+def get_point(value: str, cycling_type: Optional[str] = None) -> PointBase: ...
+
+
+@overload
+def get_point(value: None, cycling_type: Optional[str] = None) -> None: ...
+
+
 def get_point(
-    value: str, cycling_type: Optional[str] = None
+    value: Optional[str], cycling_type: Optional[str] = None
 ) -> Optional[PointBase]:
```

Comment:
This is how you type annotate a function whose return value depends on the input